src/CacheManager.StackExchange.Redis/RedisCacheHandle.cs

```diff
@@ -270,6 +270,13 @@ public override bool Exists(string key, string region)
         }
 
         /// <inheritdoc />
+        public override IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keyPattern = GetKey(pattern, region);
+            return _connection.Servers.First().Keys(_redisConfiguration.Database, keyPattern).Select(k => k.ToString());
```

Comment:
Implemented it as scanning all masters (as the slaves would be duplicates of the masters).
It'll be expensive on large clusters, but that's inevitable.